### DISCUSSION TITLE
--code arg for evaluate command?

### DISCUSSION THREAD

**Discussion by dakinggg** [explosion/spaCy#6657] [Help: Coding & Implementations] [answered]
Once I have some trained models, in order to use the `evaluate` command I think it will need some modifications. 1) a `--code` arg so that the nlp callbacks can get called. 2) a way to pass in data on the fly. For training, I have written my own functions for the `@readers` param of the corpora and I would like to use that for evaluating test data as well. 

I can of course write my own evaluate script that is mostly a copy of yours (and write the test data to disk in the correct format), but we're trying to convert over to the CLI as much as possible for v3 :) Let me know what you think the best way to proceed is, thanks!

**Comment by honnibal** [comment, marked as answer]
I would've expected to have the `--code` argument on `spacy evaluate` too, so I think this was just an oversight. Feel free to open a PR if you have time :)

**Reply by dakinggg** [reply]
I went back to look at this and realized it was just the docs that was missing the arg, the command itself does indeed have it already :) and the `develop` docs have it added
